lib/src/resources/repository.dart

```diff
@@ -0,0 +1,8 @@
+import '../models/item_model.dart';
+import 'movie_api_provider.dart';
+
+class Repository{
```

Comment:
This should conform to a generic interface for movies like `IMovieRepository` dont you think ?

Reply:
Yes, i added the interface i thinks is more easy to see how many petition we have.